common/src/View/MapDocument.cpp

```diff
@@ -1100,19 +1100,32 @@ namespace TrenchBroom {
         }
         
         bool MapDocument::csgSubtract() {
-            const Model::BrushList brushes = selectedNodes().brushes();
-            if (brushes.size() < 2)
+            const auto subtrahends = [this]() {
```

Comment:
I know this is ugly, what I'm doing here is I want to pass a `std::vector<Brush*>` to `subtract()` later on which takes a `std::vector<const Brush*>`. I could factor it out to:
```
template<class T>
std::vector<const T*> VectorUtils::copyWithConstPointers(const std::vector<T*>& vec);
```
It's wasteful at runtime, but this isn't performance critical code. Thought about the alternatives, but I don't want to move this code to headers and use iterator template arguments, also don't really want to drop const correctness (although it's difficult and this problem crops up a lot when Model::BrushList is used.)

Reply:
I have purposefully ignored const correctness in these cases because I wanted to avoid code like this. I don't know of a good solution to this problem either. Moving everything to headers and using templates seems wrong. Copying vectors around just for const correctness seems wrong. In the end, you're doing this only to document in the API that you won't be modifying the elements of the vector, which could also be done in a comment.

One can argue about such matters of style a lot, but in the end I think it's more important to be consistent than to be right. So I'd vote for disregarding the constness of the vector elements and documenting the fact that they won't be modified in some other form. Or use the badly named method `VectorUtils::cast<const Brush*>(...)` -- feel free to rename that to something that indicates that you receive a new vector, e.g. `castAndCopy` or something.